src/test/java/com/epam/edumanagementsystem/security/config/SecurityConfigTest.java

```diff
@@ -0,0 +1,133 @@
+package com.epam.edumanagementsystem.security.config;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.transaction.annotation.Transactional;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+@Transactional
+public class SecurityConfigTest {
+
```

Comment:
In this test class you've probably covered all possible redirection scenarios either positive or negative cases. If all tests are passing then this class is good. The only thing is why did you add @Transactional annotation on the class?

Reply:
There was an issue with /classses/ path due to lazy initialization, which was causing the test to fail, so it was advised to use @Transactional on the class to avoid such behavior.